apps/web/src/components/tx/ConfirmTxDetails/TxDetails.tsx

```diff
@@ -185,3 +195,27 @@ export const TxDetails = ({ safeTx, txData, showHashes }: TxDetailsProps) => {
     </PaperViewToggle>
   )
 }
+
+function TxDetailsHashes({ safeTx }: Pick<TxDetailsProps, 'safeTx'>) {
+  const safeTxHash = useSafeTxHash({ safeTxData: safeTx.data })
+  const domainHash = useDomainHash()
+  const messageHash = useMessageHash({ safeTxData: safeTx.data })
+
+  return (
+    <>
+      {[
+        ['Domain hash', domainHash] as const,
+        ['Message hash', messageHash] as const,
+        ['safeTxHash', safeTxHash] as const,
+      ].map(([label, hash]) => {
+        return (
+          <TxDetailsRow label={label} key={hash}>
+            <Typography variant="body2" width="100%" sx={{ wordWrap: 'break-word' }}>
+              {hash}
+            </Typography>
```

Comment:
I would keep the HexEncodedData component here.

Reply:
Added in 8088f81565c50bbda19222d8b920bd696b365acd.